### ISSUE TITLE
WaoN music Transcriber:   Need Help!

### ISSUE THREAD

**Issue by vinckles** [kichiki/WaoN#6]
I am looking for a Free program, able to Transcribe mp3 or WAV audio files into sheet music notation.
I believe there is such an application called WaoN. I tried emailing their email address, given, but it bounced back - 'Mailer Demon'. I would like to communicate with someone experienced with this software with a view to learning:
1 How to operate this software?
2 How successful is it for Transcribing fingerstyle guitar solos, ie the melody; harmony and accompaniment, all played by the guitars six strings, all in one?
3 Require a link for safe Download of this software, which I believe is Free to own?
      Sourceforge seem to offer WaoN as a small file but I do not know how to open it(?)   If anyone has past experience of WaoN, please answer asap.
Regards.

**Comment by ghost** [comment]
Hi @vinckles 
I have successfully built `WaoN` then tried with some audio files, thus I think you can easily do that too 😊.
All you need to get it done is:
1. Clone this awesome repository then navigate into it
2. Install some libraries, which depend on platform of your machine
    For example: I have to run
    ```bash
    $ sudo apt-get install -y build-essential fftw3-dev libsndfile1-dev libao-dev libsamplerate-dev libncurses5-dev libgtk2.0-dev
    ```
    When trying to build `WaoN` in `ubuntu` based Docker image
3. Run `make` and you will end up with some `WaoN` stuff files like `waon`, `pv` or `gwaon`
4. DONE! Use that generated `waon` binary file to convert your favorite songs and happily watching it create midi file like this:
    <pre>
    $ ./waon -i <b>input_file.wav</b> -o <b>output_file.mid</b>
    </pre>
Note that you need to prepare `.wav` file to get it works, otherwise... Well, I don't know 😝
Good luck!!

**Comment by vinckles** [comment]
Thank you for writing.  I'm afraid I don understand any of what you are telling me. I am not technically understanding of computers:   I can visit Sourceforge;  Download the tiny WaoN file, or should that be software(?);  then  which program already in my Windows 10, 32 bit computer, do I select to open it?
   My audio file was an mp3 converted to Wav, in Mono?
I do not understand what you mean by "install some libraries...", etc
Could you please clarify what I need to do, in non technical talk.
Thank you.

**Comment by vinckles** [comment]
Hello,           I do not understand your letter.
Please explain in step by step fashion exactly what I need to do?
1   Download WaoN file from Sourceforge.
2   Next, how do I  open the WaoN file?   (When I click it a box opens saying: "What program do you want to open this file with?"   It lists internet explorer;  vlc media player;  windows media player;  notebook,  etc.... Which one should I choose, please?)
3  You mention "Library files".   Where do I get them from?  What must I do if I find them?
Tell me everything in number step order, like above, then I can try it.
My audio file is a Wav, mono.
Hoping to hear from you, soon.vinckles
    On Wednesday, 26 December 2018, 03:03:30 GMT, Tran Duc Son <notifications@github.com> wrote:  
 
 
Hi @vinckles
I have successfully built WaoN then tried with some audio files, thus I think you can easily do that too 😊 .
All you need to get it done is:
   
   - Clone this awesome repository then navigate into it
   - Install some libraries, which depend on platform of your machine   
For example: I have to run   $ sudo apt-get install -y build-essential fftw3-dev libsndfile1-dev libao-dev libsamplerate-dev libncurses5-dev libgtk2.0-devWhen trying to build WaoN in ubuntu based Docker image
   - Run make and you will end up with some WaoN stuff files like waon, pv or gwaon
   - DONE! Use that generated waon binary file to convert your favorite songs and happily watching it create midi file like this:     $ ./waon -i input_file.wav -o output_file.mid
 

Note that you need to prepare .wav file to get it works, otherwise... Well, I don't know 😝
Good luck!!

**Comment by ildar** [comment]
You seem to want prebuilt binaries (for Widows). But are you sure you'll be
able to use it with quite a restricted documentation WaoN has?

**Comment by vinckles** [comment]
No, I need lots of help to try and understand what to do and how to use WaoN.  I thought it would be a software and I could load a Wav file onto it to transcribe the audio(?)

    On Wednesday, 26 December 2018, 12:21:46 GMT, ildar <notifications@github.com> wrote:  
 
 You seem to want prebuilt binaries (for Widows). But are you sure you'll be
able to use it with quite a restricted documentation WaoN has?

**Comment by ghost** [comment]
Hi @vinckles.
Which file did you open? Currently `WaoN` contains 3 programs:
- `waon`: transcriber (wav-to-mid converter)
- `pv`: phase vocoder for time-streching and pitch-shifting
- `gwaon`: GUI for `waon` and `pv`

If you open `gwaon`, you will see something like [this](http://waon.sourceforge.net/gwaon.html)

But if you try to open `waon`, it is totally different story. Basically, `waon` is a tool which was designed for programmers and not friendly to end-users at all. Even if you click at `waon` file, no windows, no buttons, no magic things will happen. All about `waon` is just command line. So it's very hard for me to talk to you by non-technical way, and I am not familiar with Windows, but at least, you need to:
1. Open *windows command prompt*
2. Navigate to folder which contains your downloaded `waon` file
3. Type command as mentioned at https://github.com/kichiki/WaoN/blob/master/README
You should ask some technical guys (may be your friends) for helps, I believe that he/she knows how to deal with this.

**Comment by vinckles** [comment]
Hi,      Thank you for your help.  It does not seem like I will be able to try the WaoN.  It is all too complicated for me. Shame!   I had hoped it would be a straight forward software, like, say Anthemscore, but I imagine it is not(?)   There are no friends I can turn to for help with this one, so I will have to just try and forget about it.     Maybe an opensource music transcriber software might appear one day in the not too distant future.  I hope so.Regards.
    On Wednesday, 26 December 2018, 13:00:33 GMT, Tran Duc Son <notifications@github.com> wrote:  
 
 
Hi @vinckles.
Which file did you open? Currently WaoN contains 3 programs:
   
   - waon: transcriber (wav-to-mid converter)
   - pv: phase vocoder for time-streching and pitch-shifting
   - gwaon: GUI for waon and pv

If you open gwaon, you will something like this

But if you try to open waon, it is totally different story. Basically, waon is a tool which was designed for programmers and not friendly to end-users at all. Even if you click at waon file, no windows, no buttons, no magic things will happen. All about waon is just command line. So it's very hard for me to talk to you by non-technical way, and I am not familiar with Windows, but at least, you need to:
   
   - Open windows command prompt
   - Navigate to folder which contains your downloaded waon file
   - Type command as mentioned at https://github.com/kichiki/WaoN/blob/master/README   
You should ask some technical guys (may be your friends) for helps, I believe that he/she knows how to deal with this.

**Comment by ghost** [comment]
Dear @vinckles 
After spending few days keep searching around the internet, I got some interesting things for you.
You can visit this [awesome site](https://piano-scribe.glitch.me/) to transcribe piano raw audio to midi, then print it into beautiful sheet music using this [program](https://musescore.org/en).
I have not tried yet, but somehow I feel that it will smoothly work without any problem!
Best regards.

**Comment by vinckles** [comment]
Very kind of you to write.   I will try to check it out.  Not sure what I should be clicking on to open up an online trascribing device(?)    My music is guitar,  in mp3 and  wav file type.  Hope it can accept them, but as I say, so far I have not found how to import them(?)   These kind of sites always leave you with a mystery!
    On Monday, 7 January 2019, 01:22:03 GMT, Tran Duc Son <notifications@github.com> wrote:  
 
 
Dear @vinckles
After spending few days keep searching around the internet, I got some interesting things for you.
You can visit this awesome site to transcribe piano raw audio to midi, then print it into beautiful sheet music using this program
I have not tried yet, but somehow I feel that it will smoothly works.
Best regards.

**Comment by realfinder** [comment]
windows build with msys2 mingw
[waon.zip](https://github.com/kichiki/WaoN/files/6958228/waon.zip)
